appium-dotnet-driver/Appium/AppiumDriver.cs

```diff
@@ -773,27 +580,61 @@ public void Zoom(IWebElement el)
         /// Gets device date and time for both iOS(Supports only real device) and Android devices
         /// </summary>
         /// <returns>A string which consists of date and time</returns>
-        public string DeviceTime
-        {
-            get { return ((IExecuteMethod) this).Execute(AppiumDriverCommand.GetDeviceTime).Value.ToString(); }
-        }
+        public string DeviceTime => ((IExecuteMethod) this).Execute(AppiumDriverCommand.GetDeviceTime).Value.ToString();
 
         #endregion Device Time
 
         #region Session Data
 
-        /// <summary>
-        /// This property returns a dictionary of the current session data
-        /// </summary>
-        public Dictionary<string, object> SessionDetails
+        public IDictionary<string, object> SessionDetails
         {
             get
             {
-                return (Dictionary<string, object>) ((IExecuteMethod) this).Execute(AppiumDriverCommand.GetSession)
+                var session = 
+                    (IDictionary<string, object>) ((IExecuteMethod) this).Execute(AppiumDriverCommand.GetSession)
                     .Value;
+                return new ReadOnlyDictionary<string, object>(session.Where(entry =>
+                {
+                    var key = entry.Key;
+                    var value = entry.Value;
+                    return !string.IsNullOrEmpty(key) &&
+                           value != null && !string.IsNullOrEmpty(Convert.ToString(value));
+                }).ToDictionary(entry => entry.Key, entry => entry.Value));
             }
         }
 
+        public object GetSessionDetail(string detail)
+        {
+            var details = SessionDetails;
+            return details.ContainsKey(detail) ? details[detail] : null;
+        }
+
+        public string PlatformName
+        {
+            get
+            {
+                var platform = GetSessionDetail("platformName");
+                if (platform == null || string.IsNullOrEmpty(Convert.ToString(platform)))
+                {
+                    platform = GetSessionDetail("platform");
+                }
+
+                return platform == null ? null : Convert.ToString(platform);
+            }
+        }
+
+        public string AutomationName
+        {
+            get
+            {
+                var automation = GetSessionDetail("automationName");
+                return automation == null ? null : Convert.ToString(automation);
+            }
+        }
+
+        public bool IsBrowser => GetSessionDetail("browserName") != null
+                                 && Context.IndexOf("NATIVE_APP", StringComparison.OrdinalIgnoreCase) < 0;

```

Comment:
for both "browerName" and "NATIVE_APP"  can use their "static readonly" in their respective files.